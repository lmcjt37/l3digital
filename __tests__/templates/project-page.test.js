import { graphql, StaticQuery } from 'gatsby'

import ProjectPage from 'templates/project-page'
import ProjectPageMock from '__mocks__/templates/project-page.mock'
import SEOMock from '__mocks__/components/SEO.mock'

beforeEach(() => {
    graphql.mockImplementation(() => ProjectPageMock)
    // Mock SEO component to prevent Invariant errors
    StaticQuery.mockImplementationOnce(({ render }) => render(SEOMock))
})

describe('Project Page template tests', () => {
    test('check it renders correctly with snapshot', () => {
        const tree = render(<ProjectPage {...ProjectPageMock} />)
        expect(tree).toMatchSnapshot()
    })
})
