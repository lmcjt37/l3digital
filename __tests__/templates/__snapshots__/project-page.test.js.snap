// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Page template tests check it renders correctly with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="alt"
          id="main"
        >
          <section
            id="one"
          >
            <div
              class="inner"
            >
              <a
                class="button previous-effect effect bottom-margin"
                href="/projects"
                partiallyactive="false"
                state="[object Object]"
              >
                Back
              </a>
              <header
                class="major"
              >
                <h1>
                  Project page title
                </h1>
              </header>
              <span
                class="image left"
              >
                <div
                  class=" gatsby-image-wrapper"
                  style="position: relative; overflow: hidden;"
                >
                  <div
                    style="width: 100%; padding-bottom: 100%;"
                  />
                  <img
                    alt=""
                    src="base64"
                    style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition: opacity 0.5s; transition-delay: 0.25s;"
                  />
                  <picture>
                    <source
                      sizes="sizes"
                      srcset="srcSet"
                    />
                    <img
                      alt="featured image description"
                      src="src"
                      style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 0.5s;"
                    />
                  </picture>
                  <noscript>
                    &lt;picture&gt;&lt;source srcSet="srcSet" sizes="sizes" /&gt;&lt;img src="src" alt="featured image description" style="position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
                  </noscript>
                </div>
              </span>
              <div
                class="text right"
              >
                <p>
                  Project page description
                </p>
                <p />
              </div>
              <ul
                class="actions clearfix"
              >
                <li>
                  <a
                    class="button special"
                    href="http://l3digital.co.uk"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Visit website
                  </a>
                </li>
              </ul>
            </div>
          </section>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="alt"
        id="main"
      >
        <section
          id="one"
        >
          <div
            class="inner"
          >
            <a
              class="button previous-effect effect bottom-margin"
              href="/projects"
              partiallyactive="false"
              state="[object Object]"
            >
              Back
            </a>
            <header
              class="major"
            >
              <h1>
                Project page title
              </h1>
            </header>
            <span
              class="image left"
            >
              <div
                class=" gatsby-image-wrapper"
                style="position: relative; overflow: hidden;"
              >
                <div
                  style="width: 100%; padding-bottom: 100%;"
                />
                <img
                  alt=""
                  src="base64"
                  style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition: opacity 0.5s; transition-delay: 0.25s;"
                />
                <picture>
                  <source
                    sizes="sizes"
                    srcset="srcSet"
                  />
                  <img
                    alt="featured image description"
                    src="src"
                    style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 0.5s;"
                  />
                </picture>
                <noscript>
                  &lt;picture&gt;&lt;source srcSet="srcSet" sizes="sizes" /&gt;&lt;img src="src" alt="featured image description" style="position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
                </noscript>
              </div>
            </span>
            <div
              class="text right"
            >
              <p>
                Project page description
              </p>
              <p />
            </div>
            <ul
              class="actions clearfix"
            >
              <li>
                <a
                  class="button special"
                  href="http://l3digital.co.uk"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Visit website
                </a>
              </li>
            </ul>
          </div>
        </section>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
